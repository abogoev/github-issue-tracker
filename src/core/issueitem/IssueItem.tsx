import React from "react";
import { VFC } from "react";
import { StyleSheet, Text, View } from "react-native";
import OpenIssueIcon from "../../../assets/svg/open-issue.svg";
import ClosedIssueIcon from "../../../assets/svg/closed-issue.svg";
import theme from "../../theme/theme";
import CustomTouchableOpacity, {
  CustomTouchableOpacityProps,
} from "../touchableopacity/CustomTouchableOpacity";
// import Bookmark from "../bookmark/Bookmark";
import { WithoutChildren } from "../../types";

interface Props extends WithoutChildren<CustomTouchableOpacityProps> {
  title: string;
  createdAt: Date;
  closedAt: Date | null;
  user: string;
  number: number;
}

const IssueItem: VFC<Props> = ({
  style,
  title,
  createdAt,
  closedAt,
  user,
  number,
  ...props
}) => {
  return (
    <CustomTouchableOpacity style={[styles.container, style]} {...props}>
      <View style={styles.topContainer}>
        {closedAt ? (
          <ClosedIssueIcon style={styles.icon} />
        ) : (
          <OpenIssueIcon style={styles.icon} />
        )}
        <View style={styles.textContainer}>
          <Text style={styles.title} numberOfLines={1}>
            {title}
          </Text>
          <Text
            numberOfLines={1}
            style={styles.description}
          >{`#${number} opened on ${createdAt.toLocaleString()} by ${user}`}</Text>
        </View>
        {/* <Bookmark owner={owner} repo={repo} number={number} /> */}
      </View>
    </CustomTouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 8,
    backgroundColor: theme.palette.white,
    borderColor: theme.palette.neutral,
    borderWidth: 1,
    minHeight: 62,
  },
  topContainer: {
    flexDirection: "row",
  },
  icon: {
    marginHorizontal: 8,
  },
  textContainer: {
    flex: 1,
  },
  title: {
    ...theme.typography.largeText,
    color: theme.palette.veryDarkGrey,
  },
  description: {
    ...theme.typography.smallText,
    marginTop: 4,
    color: theme.palette.darkGrey,
  },
});

export default IssueItem;
