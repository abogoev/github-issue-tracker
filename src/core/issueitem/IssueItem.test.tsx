import React from "react";
import IssueItem from "./IssueItem";
import { create } from "react-test-renderer";

describe("OpenClosedBadge", () => {
  it("should render correctly", () => {
    const mockOnPress = jest.fn();

    const component = create(
      <IssueItem
        title="Hello Lorem ipsum dolor sit amet consectetur adipisicing elit. Qui corrupti dolorem est eius cum tempora id "
        number={5}
        createdAt={new Date("2000-01-01")}
        closedAt={null}
        user={{ avatar_url: "", login: "vanko" }}
        onPress={mockOnPress}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
