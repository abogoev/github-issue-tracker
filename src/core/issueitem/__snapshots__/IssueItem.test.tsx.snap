// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenClosedBadge should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "rgb(208, 215, 222)",
      "borderWidth": 1,
      "minHeight": 62,
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <svg
      data-file-name="SvgOpenIssue"
      style={
        Object {
          "marginHorizontal": 8,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#24292f",
            "fontFamily": "Helvetica",
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          }
        }
      >
        Hello Lorem ipsum dolor sit amet consectetur adipisicing elit. Qui corrupti dolorem est eius cum tempora id 
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "rgb(87, 96, 106)",
            "fontFamily": "Helvetica",
            "fontSize": 12,
            "lineHeight": 18,
            "marginTop": 4,
          }
        }
      >
        #5 opened on 01/01/2000, 02:00:00 by vanko
      </Text>
    </View>
  </View>
</View>
`;
